todo-app/src/App.js

```diff
@@ -1,10 +1,86 @@
 import React, { Component } from 'react';
+import Header from './Components/Header'
+import Footer from './Components/Footer'
+import Container from './Components/Container'
+
+
+import './App.css';
+const data = require('./todoList.json');
 
 class App extends Component {
+  state = {
+    todoItems: []
+  }
+  componentWillMount() {
+    localStorage.getItem('todoItems') && this.setState({
+      todoItems: JSON.parse(localStorage.getItem('todoItems'))
+    })
+  }
+  componentDidMount() {
+    if (!localStorage.getItem('todoItems')) {
+      this.setState({
+        todoItems: data
+      })
+    } else {
+      console.log('we are using the data from the Local Storage.')
+    }
+
+  }
+  componentWillUpdate(nextProps, nextState) {
+    localStorage.setItem('todoItems', JSON.stringify(nextState.todoItems))
+  }
+
+  handelSelected = (index) => {
+    const todoItems = JSON.parse(localStorage.getItem('todoItems'))
+    todoItems[index].selected = !todoItems[index].selected
+    this.setState({ todoItems })
+  }
+
+  handelNewTask = (newTask, newDeadLine) => {
+
+    const todoItems = this.state.todoItems
+    todoItems.push({
+      "id": this.state.todoItems.length + 1,
+      "description": newTask,
+      "deadline": newDeadLine,
+      "selected": false
+    })
+    this.setState({ todoItems })
+  }
+
+  handelUpdateTask = (index , update) => {
```

Comment:
Again, please use ID to locate items inside an array, not index. You're already doing this for `handelDeleteTask`, just extend the functionality here.